mir_eval/segment.py

```diff
@@ -1178,6 +1137,35 @@ def vmeasure(
         marginal=True,
     )
 
+def return_mapping(mapping:Sequence[Any], some_func, *args, **kwargs) -> dict[Any, Any]:
+    """Return function results as a mapping
+    
+    Note: there are no type-checks or length-checks in this method, clients are responsible.
+
+    Args:
+        mapping (Sequence): keys to form mapping, same order as return values of some_func
+        some_func (_type_): function to decorate
+
+    Returns:
+        dict: return values of some_func, mapped by mapping key
+    """
+    return_res = some_func(*args, **kwargs)
+    return _return_values_to_mapping(mapping, return_res)
+
+def _return_values_to_mapping(mapping:Sequence[Any], values, accumulator:dict=None) -> dict[Any, Any]:
+    if accumulator is None:
+        accumulator = {}
+    if not isinstance(values, tuple):
+        accumulator[mapping[0]] = values
+    else:
+        for i,m in enumerate(mapping):
+            accumulator[m] = values[i]
+    return accumulator
```

Comment:
I don't understand what these functions do - it seems like abstraction that may not be necessary, and it definitely does not help code readability.